### ISSUE TITLE
FK differences between 2 tables. 

### ISSUE THREAD

**Issue by WillEhrendreich** [dotnet/efcore#28203]
What do I do about these columns being different in SQLServer?

 I've got a main table that has int as it's primary key. it is generated on add. THEN i have an "extension" table, which was made by someone else and I'm unable to change it, but it's foreign key column that points to the main table is a bigint. 

if i change the clr representation from int64 to int, and then give it a .HasConversion<int64> in the configuration, i get THIS exception when i try to save: 

InvalidCastException: Unable to cast object of type 'Microsoft.EntityFrameworkCore.Update.Internal.SimpleRowKeyValueFactory`1[System.Int32]' to type 'Microsoft.EntityFrameworkCore.Update.Internal.IRowKeyValueFactory`1[System.Int64]'.

if i keep the clr representation of the extension fk to int64, then i get this:

The relationship from 'BomRowExtension' to 'BomRow.Ext' with foreign key properties {'MainRowKey' : long} cannot target the primary key {'IdentityColumn' : int} because it is not compatible. Configure a principal key or a set of foreign key properties with compatible types for this relationship.

this seems like a damned if you do damned if you don't thing, but i swear I've had this working before, i just don't know how i changed it so it doesn't work now.

**Comment by ajcvickers** [comment]
@WillEhrendreich A value converter should work in this case--see my code below. Please attach a small, runnable project or post a small, runnable code listing that reproduces what you are seeing so that we can investigate.

```C#
public class Blog
{
    public int Id { get; set; }
    public string Title { get; set; }

    public ICollection<Post> Posts { get; set; } = new List<Post>();
}

public class Post
{
    public int Id { get; set; }

    [Column(TypeName = "bigint")]
    public int? BlogId { get; set; }

    public Blog Blog { get; set; }
}

public class SomeDbContext : DbContext
{
    protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
        => optionsBuilder
            .UseSqlServer(Your.ConnectionString)
            .LogTo(Console.WriteLine, LogLevel.Information)
            .EnableSensitiveDataLogging();

    
    public DbSet<Blog> Blogs { get; set; }
    public DbSet<Post> Posts { get; set; }
}

public class Program
{
    public static void Main()
    {
        using (var context = new SomeDbContext())
        {
            context.Add(new Blog {Posts = {new Post(), new Post()}});
            context.SaveChanges();
        }
        
        using (var context = new SomeDbContext())
        {
            var blogs = context.Blogs.Include(e => e.Posts).ToList();
            foreach (var blog in blogs)
            {
                Console.WriteLine(blog.Posts.Count);
            }
        }
    }
}
```

**Comment by WillEhrendreich** [comment]
> @WillEhrendreich A value converter should work in this case--see my code below. Please attach a small, runnable project or post a small, runnable code listing that reproduces what you are seeing so that we can investigate.
> 
> ```C#
> public class Blog
> {
>     public int Id { get; set; }
>     public string Title { get; set; }
> 
>     public ICollection<Post> Posts { get; set; } = new List<Post>();
> }
> 
> public class Post
> {
>     public int Id { get; set; }
> 
>     [Column(TypeName = "bigint")]
>     public int? BlogId { get; set; }
> 
>     public Blog Blog { get; set; }
> }
> 
> public class SomeDbContext : DbContext
> {
>     protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
>         => optionsBuilder
>             .UseSqlServer(Your.ConnectionString)
>             .LogTo(Console.WriteLine, LogLevel.Information)
>             .EnableSensitiveDataLogging();
> 
>     
>     public DbSet<Blog> Blogs { get; set; }
>     public DbSet<Post> Posts { get; set; }
> }
> 
> public class Program
> {
>     public static void Main()
>     {
>         using (var context = new SomeDbContext())
>         {
>             context.Add(new Blog {Posts = {new Post(), new Post()}});
>             context.SaveChanges();
>         }
>         
>         using (var context = new SomeDbContext())
>         {
>             var blogs = context.Blogs.Include(e => e.Posts).ToList();
>             foreach (var blog in blogs)
>             {
>                 Console.WriteLine(blog.Posts.Count);
>             }
>         }
>     }
> }
> ```

That's what I'm saying, I put a value conversion on it, and I get an error when I never used to. I give the property a .HasConversion<int64>() and it fails, and I swear it never used to.

The only difference I can see between this and mt old version that worked is that I rewrote it in fsharp. I can't see why that would make it fail though.

**Comment by ajcvickers** [comment]
**EF Team Triage:** Closing this issue as the requested additional details have not been provided and we have been unable to reproduce it.

*BTW this is a canned response and may have info or details that do not directly apply to this particular issue. While we'd like to spend the time to uniquely address every incoming issue, we get a lot traffic on the EF projects and that is not practical. To ensure we maximize the time we have to work on fixing bugs, implementing new features, etc. we use canned responses for common triage decisions.*

**Comment by WillEhrendreich** [comment]
If anyone is interested, I gave up this approach. I just wrote it in csharp again with primitive types, and any type safety I enforced was between the DTO objects and the domain model, not between the DTO's and EFCore. EFCore did not seem to enjoy me trying to give it fsharp types. Not AS nice, but still workable and gets the goals accomplished. Cheers to anyone who can figure out how to make it work the other way.
